docs/cloud-functions/gse-roar-admin/sync-assignment-deleted.md

```diff
@@ -0,0 +1,66 @@
+  
+- **Transaction Execution**:
+  - **Retrieval of Previous Data**: Fetches data from the deleted document to guide updates.
+  - **Metric Adjustments**: Utilizes helper functions to decrement status counters in the `administrations/stats` collection based on the data from the deleted document.
+  - **User Document Update**: Adjusts the user's document to reflect the removal of the assignment, ensuring that any reference to the deleted assignment is cleared from tracking fields.
+
+### Error Handling and Security Considerations
+
+- **Error Handling**:
+  - Implements robust handling of transaction failures and ensures that all operations are retried up to a limit if they fail initially.
+  - Manages potential inconsistencies that may arise if the deletion event does not provide complete data.
+
+- **Security**:
+  - Ensures that operations are only triggered by authorized deletions, with checks to prevent unauthorized data manipulation.
+  - Validates that only existing and accurate data is used to update counters and user profiles, ensuring integrity in data processing.
+
+### Deployment and Maintenance
+
+- **Deployment**: This function should be deployed using Firebase CLI tools or through the Firebase Console. It is crucial to ensure that the function is correctly configured to listen to the intended document paths and that it has the necessary permissions to execute read and write operations on the database.
+
+- **Maintenance**: Requires ongoing monitoring to ensure it performs as expected without leading to data inconsistencies or failures. Logs should be reviewed regularly, and the function should be updated to adapt to any changes in the underlying data schema or business logic.
+
+
+[//]: # (This function will trigger when an assignment is deleted in a user's assignments subcollection. It serves to update all the places where the assignment is being listed and counted, namely the user's assignments object in the `/users/{userId}` document and the administration's stats document.)
+
+[//]: # ()
+[//]: # (When an assignment is deleted, the function will update the user's `assignmentsAssigned` object to remove the assignment ID if found.)
+
+[//]: # ()
+[//]: # (The function will also update the administration's stats document at `/administrations/{administrationId}/stats/completion` to no longer count this student. This document keeps track of the number of students who have have been assigned, started, and completed the assignment. This function will update both the total number of students assigned, as well as the number of students from the user's particular organization that have been assigned the administration.)
+
+[//]: # ()
+[//]: # (::: tip)
+
+[//]: # (The document IDs in the `/users/{userId}/assignments` collection will always match with the administration they refer to, so when a new file is created we can assume that it is a new assignment to the user.)
+
+[//]: # (:::)
```

Comment:
Same here. This seems redundant to the above documentation.

Reply:
This was Elijah's original wording, I kept it in the repo as a comment in case we wanted to include it in some way later on.